src/hotspot/cpu/x86/gc/z/z_x86_64.ad

```diff
@@ -137,12 +120,13 @@ instruct zCompareAndSwapP(rRegI res, memory mem, rRegP newval, rRegP tmp, rFlags
             "movzbl   $res, $res" %}
 
   ins_encode %{
-    if (barrier_data() != ZLoadBarrierElided) {
+    if (barrier_data() != 0) { // barrier could be elided by ZBarrierSetC2::analyze_dominating_barriers()
```

Comment:
The information that there was a barrier attached, is implicit in the ins_encode block due to it being run at all. In other words, since we matched the mach node to our ZGC access instead of a normal access, we already know that there was barrier data attached, and that we no longer have such barrier data.

Reply:
Ok, makes sense. What do you think about making `ZLoadBarrierElided = 0` then?

Reply:
I'm okay with that. I don't have a strong preference.

Reply:
I also prefer to have ZLoadBarrierElided = 0. I will add it.